src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,326 @@
+        :param create_index: If true, creates a new vector search index.
+        :param index_type: Vector search index type ("FLAT", "IVF_FLAT").
+        :param index_creation_kwargs: Arguments to be passed to the index creation
+            method.
+        # DirectoryTextReader params.
+        :param glob: Glob pattern relative to the specified path by default set to
+            pick up all non-hidden files.
+        :param exclude: Patterns to exclude from results, use glob syntax.
+        :param suffixes: Load only files with these suffixes. Suffixes must include
+            the dot, e.g. ".txt".
+        :param text_splitter_kwargs: Arguments for the splitter class.
+        # SentenceTransformersEmbedding params.
+        :param model_name_or_path: Huggingface SentenceTransformer model name or path
+        # Index update params.
+        :param index_timestamp: Timestamp to add index updates at.
+        :param workers: If `embeddings_generation_mode=BATCH` this is the number of
+            distributed workers to be used.
+        :param worker_resources: If `embeddings_generation_mode=BATCH` this can be used
+            to specify the worker resources.
+        :param worker_image: If `embeddings_generation_mode=BATCH` this can be used
+            to specify the worker Docker image.
+        :param extra_worker_modules: If `embeddings_generation_mode=BATCH` this can be
+            used to install extra pip package to the image.
+        :param driver_resources: If `embeddings_generation_driver_mode=BATCH` this can
+            be used to specify the driver resources.
+        :param driver_image: If `embeddings_generation_driver_mode=BATCH` this can be
+            used to specify the driver Docker image.
+        :param extra_driver_modules: If `embeddings_generation_driver_mode=BATCH` this
+            can be used to install extra pip package to the image.
+        :param max_tasks_per_stage: Number of maximum udf tasks per computation stage.
+        :param embeddings_generation_mode: TaskGraph execution mode for embeddings
+            generation.
+        :param embeddings_generation_driver_mode: TaskGraph execution mode for the
+            ingestion driver.
+        :param vector_indexing_mode: TaskGraph execution mode for the vector indexing.
+        :param index_update_kwargs: Extra arguments to pass to the index update job.
+        """
+        import tiledb
+
+        # Install extra packages not yet available in vectorsearch UDF image.
+        # TODO(nikos) remove the pacakge installations after the image is updated
+        def install_extra_worker_modules():
+            import os
+            import subprocess
+            import sys
+
+            sys.path.insert(0, "/home/udf/.local/bin")
+            sys.path.insert(0, "/home/udf/.local/lib/python3.9/site-packages")
+            os.environ["PATH"] = f"/home/udf/.local/bin:{os.environ['PATH']}"
+            subprocess.check_call(
```

Comment:
This needs to be removed. Add packages required to default images.

Reply:
This is just a temporary solution to test before the release of the new UDF image. If we remove this the PR will need to block on the UDF image release.

Reply:
We cannot ship as is, this is a blocking change request. Please add the packages needed to the default UDF images.

Reply:
Removed